packages/helper-backend/src/deepchat-proxy/deepchat-proxy.controller.ts

```diff
@@ -45,16 +119,127 @@ export class DeepchatProxyController {
   }
 
   @Post()
-  async create(@Body() mapping: DeepchatProxy): Promise<DeepchatProxy> {
+  @ApiOperation({
+    summary: 'Create a new proxy',
+    description: 'Creates a new Deepchat proxy with the provided configuration details'
+  })
+  @ApiBody({
+    type: CreateProxyMappingDto,
+    description: 'Deepchat proxy configuration',
+    schema: {
+      example: {
+        model: 'gpt-4',
+        url: 'https://api.openai.com/v1',
+        apiKey: 'sk-...'
+      }
+    }
+  })
+  @ApiResponse({
+    status: 201,
+    description: 'The proxy has been successfully created',
+    type: DeepchatProxy,
+    schema: {
+      properties: {
+        _id: { type: 'string', example: '6401234567890abcdef12345' },
+        model: { type: 'string', example: 'claude-3-opus' },
+        url: { type: 'string', example: 'https://api.anthropic.com/v1' }
+      }
+    }
+  })
+  @ApiResponse({ status: 400, description: 'Invalid input data or model already exists' })
+  async create(@Body() mapping: CreateProxyMappingDto): Promise<DeepchatProxy> {
     return this.deepchatProxyService.create(mapping);
   }
 
   @Delete('/:model')
+  @ApiOperation({
+    summary: 'Delete a proxy',
+    description: 'Deletes a Deepchat proxy by model name'
+  })
+  @ApiParam({
+    name: 'model',
+    description: 'Model name to delete',
+    example: 'gpt-4'
+  })
+  @ApiResponse({ status: 200, description: 'The proxy has been successfully deleted' })
+  @ApiResponse({ status: 404, description: 'Proxy not found' })
   async delete(@Param('model') model: string): Promise<void> {
     await this.deepchatProxyService.delete(model);
   }
 
   @Post('proxy/:id')
+  @ApiOperation({
+    summary: 'Proxy a request',
+    description: 'Proxies a completion request to the specified LLM provider using the stored configuration'
+  })
+  @ApiParam({
+    name: 'id',
+    description: 'Proxy identifier (MongoDB ObjectId)',
+    example: '6401234567890abcdef12345'
+  })
+  @ApiBody({
```

Comment:
You can remove the `ApiBody` and schema on `ApiResponse` where the class in question already has `ApiProperty` tags.